test/framework/options.py

```diff
@@ -2970,6 +2970,33 @@ def test_stop(self):
         regex = re.compile("COMPLETED: Installation STOPPED successfully", re.M)
         self.assertTrue(regex.search(txt), "Pattern '%s' found in: %s" % (regex.pattern, txt))
 
+    def test_fetch(self):
+        options = EasyBuildOptions(go_args=['--fetch'])
+
+        self.assertTrue(options.options.fetch)
+        self.assertEqual(options.options.stop, 'fetch')
+        self.assertEqual(options.options.modules_tool, None)
+        self.assertTrue(options.options.ignore_osdeps)
+
+        # in this test we want to fake the case were no modules tool are in the system so teak it
+        original_modtool = self.modtool
+        self.modtool = None
+        args = ['toy-0.0.eb', '--fetch']
+        stdout, stderr = self._run_mock_eb(args, raise_error=True, strip=True, testing=False)
+        # back to orinal modules tool so further tests can continue
+        self.modtool = original_modtool
```

Comment:
@MisterFruits There's no need to restore `self.modtool`, it gets reset for every test automatically.

The tests are pretty well isolated from each other.